ktor-utils/common/src/io/ktor/util/pipeline/StackTraceRecover.kt

```diff
@@ -7,8 +7,9 @@ package io.ktor.util.pipeline
 import kotlin.coroutines.*
 import kotlinx.coroutines.internal.*
 
-
 internal fun recoverStackTraceBridge(exception: Throwable, continuation: Continuation<*>): Throwable {
     @Suppress("INVISIBLE_MEMBER")
-    return recoverStackTrace(exception, continuation)
+    return recoverStackTrace(exception, continuation).withCause(exception.cause)
 }
```

Comment:
Here you rely on possibly binary incompatible API. `try {} catch (e: LinkageError) { /* ignore */ }` should do the trick

Reply:
Definitely! Probably we should ignore any `Throwable` here.